### PR TITLE
Add regex support

### PR DESCRIPTION
**Pull request by madhawa-gunasekara**

### CODE REVIEW

File: modules/ballerina-native/src/main/java/org/ballerinalang/nativeimpl/lang/regex/FindAll.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+ * @since 0.90
+ */
+@BallerinaFunction(
+        packageName = "ballerina.lang.regex",
+        functionName = "findAll",
+        args = {@Argument(name = "regex", type = TypeEnum.STRING),
+                @Argument(name = "inputText", type = TypeEnum.STRING)},
+        returnType = {@ReturnType(type = TypeEnum.ARRAY, elementType = TypeEnum.STRING)},
+        isPublic = true)
+@BallerinaAnnotation(annotationName = "Description", attributes = {@Attribute(name = "value",
```
Comment (lankavitharana) [inline modules/ballerina-native/src/main/java/org/ballerinalang/nativeimpl/lang/regex/FindAll.java]: Shall we remove these? these are already there in the native.bal

File: modules/ballerina-native/src/main/ballerina/ballerina/lang/regex/natives.bal [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+@doc:Param { value:"inputText: The string to be applied the regular expression" }
+@doc:Return { value:"boolean: True if the entire string match the regular expression; false otherwise" }
+native function match (string regex, string inputText) (boolean);
+
+@doc:Description { value:"Replaces every sub sequence of the input string that matches the against the regular expression with the replacingText"}
+@doc:Param { value:"inputText: The string to be applied the regular expression" }
+@doc:Param { value:"regex: Regular expression to be applied" }
+@doc:Param { value:"replacingText: The replacement string" }
+@doc:Return { value:"string: The derived string" }
+native function replaceAll (string inputText, string regex, string replacingText) (string);
```
Comment (SupunS) [inline modules/ballerina-native/src/main/ballerina/ballerina/lang/regex/natives.bal]: Other two functions take the regex as the first input. Shall we reorder the input params of this function, to be aligned with those two (or the vise-versa)?

**Comment by hasithaa** [comment]
@madhawa-gunasekara Can you update this PR with the latest master and resolve conflicts.

**Comment by afkham** [comment]
@madhawa-gunasekara we have made some changes to the string type so it is not possible to merge this PR. However, we think that this will be useful to have. Can you fix and resend the PR?

**Comment by sameera-jayasoma** [comment]
We are closing this PR. Please send a new one

**Pull request closed without merging**